src/components/tl/nccl/tl_nccl_coll.c

```diff
@@ -167,21 +168,21 @@ ucc_status_t ucc_tl_nccl_alltoallv_start(ucc_coll_task_t *coll_task)
 exit_coll:
     if (status == UCC_OK) {
         ucc_progress_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
-    } else if (status < 0) {
+    } else if (ucc_unlikely(status < 0)) {
         task->super.super.status = status;
     }
     return status;
 }
 
 ucc_status_t ucc_tl_nccl_alltoallv_init(ucc_tl_nccl_task_t *task)
 {
-    if (UCC_IS_INPLACE(task->args)) {
+    if (ucc_unlikely(UCC_IS_INPLACE(task->args))) {
         tl_error(UCC_TL_TEAM_LIB(task->team),
                  "inplace alltoall is not supported");
         return UCC_ERR_NOT_SUPPORTED;
     }
-    if ((task->args.src.info_v.datatype == UCC_DT_USERDEFINED) ||
-        (task->args.dst.info_v.datatype == UCC_DT_USERDEFINED)) {
+    if (ucc_unlikely((task->args.src.info_v.datatype == UCC_DT_USERDEFINED) ||
```

Comment:
the same applies to all the checks for userdefined type or reduction op: we don't know if it is likely or not for some random application. Your logic is: most apps don't use userdefined types, so this is unlikely. But this means that for the app that will use it - it will always be wrong. So, i think we should not add unlikely for such cases.